irksome/galerkin_stepper.py

```diff
@@ -142,35 +158,34 @@ def getFormGalerkin(F, L_trial, L_test, Qdefault, t, dt, u0, stages, bcs=None, a
     Fnew = sum(getTermGalerkin(Fcur, L_trial, L_test, Q, t, dt, u0, stages, test, aux_indices)
                for Q, Fcur in splitting.items())
 
-    # mass-ish matrix for BC, based on default quadrature rule
-    qpts = Qdefault.get_points()
-    qwts = Qdefault.get_weights()
-    tabulate_trials = L_trial.tabulate(0, qpts)
-    trial_vals = tabulate_trials[(0,)]
-    test_vals = L_test.tabulate(0, qpts)[(0,)]
-    test_vals_w = np.multiply(test_vals, qwts)
-
-    mmat = test_vals_w @ trial_vals[1:].T
-
-    proj = vecconst(np.linalg.solve(mmat, test_vals_w))
-    trial_vals = vecconst(trial_vals)
-    qpts = vecconst(qpts.reshape((-1,)))
-
     # Oh, honey, is it the boundary conditions?
+    i0, = L_trial.entity_dofs()[0][0]
+    nodes = list(L_trial.dual_basis())
+    del nodes[i0]
+    # list of dictionaries mapping time coordinates to weights to evaluate DOFs
+    trial_dicts = [{Constant(c): Constant(sum(w for (w, *_) in wts))
+                   for (c,), wts in node.pt_dict.items()} for node in nodes]
+    dtrial_dicts = [{Constant(c): Constant(sum(w for (w, *_) in wts))
+                    for (c,), wts in node.deriv_dict.items()} for node in nodes]
+    aux_dicts = [{Constant(c): Constant(sum(w for (w, *_) in wts))
+                 for (c,), wts in node.pt_dict.items()} for node in L_test.dual_basis()]
+
     V = u0.function_space()
     if bcs is None:
         bcs = []
     bcsnew = []
     for bc in bcs:
-        u0_sub = bc2space(bc, u0)
         g0 = as_ufl(bc._original_arg)
-        Vg_np = np.array([replace(g0, {t: t + c * dt}) for c in qpts])
-        Vg_np -= u0_sub * trial_vals[0]
-        g_np = proj @ Vg_np
+        dtg0 = expand_time_derivatives(TimeDerivative(g0), t=t, timedep_coeffs=(u0,))
         for i in range(num_stages):
+            # Evaluate the degrees of freedom
```

Comment:
I redid the BCs because the previous code relied on the default quadrature, which could potentially be under-integrating the L2 projection for Lobatto. These new BCs correspond to `bc_type="ODE"`, but we can easily restore the old ones as `bc_type="DAE"`